tools/shell/linenoise.cpp

```diff
@@ -139,25 +139,24 @@ static int history_len = 0;
 static char **history = NULL;
 static char *history_file = NULL;
 #ifndef DISABLE_HIGHLIGHT
-#include <string>
 
 static int enableHighlighting = 1;
 struct Color {
 	const char *color_name;
 	const char *highlight;
 };
 static Color terminal_colors[] = {{"red", "\033[31m"},           {"green", "\033[32m"},
-                                  {"yellow", "\033[33m"},        {"blue", "\033[34m"},
-                                  {"magenta", "\033[35m"},       {"cyan", "\033[36m"},
-                                  {"white", "\033[37m"},         {"brightblack", "\033[90m"},
-                                  {"brightred", "\033[91m"},     {"brightgreen", "\033[92m"},
-                                  {"brightyellow", "\033[93m"},  {"brightblue", "\033[94m"},
-                                  {"brightmagenta", "\033[95m"}, {"brightcyan", "\033[96m"},
-                                  {"brightwhite", "\033[97m"},   {nullptr, nullptr}};
-static std::string bold = "\033[1m";
-static std::string keyword = "\033[32m\033[1m";
-static std::string constant = "\033[33m";
-static std::string reset = "\033[00m";
+								{"yellow", "\033[33m"},        {"blue", "\033[34m"},
+								{"magenta", "\033[35m"},       {"cyan", "\033[36m"},
+								{"white", "\033[37m"},         {"brightblack", "\033[90m"},
+								{"brightred", "\033[91m"},     {"brightgreen", "\033[92m"},
+								{"brightyellow", "\033[93m"},  {"brightblue", "\033[94m"},
+								{"brightmagenta", "\033[95m"}, {"brightcyan", "\033[96m"},
+								{"brightwhite", "\033[97m"},   {nullptr, nullptr}};
+char const *bold = "\033[1m";
```

Comment:
One caveat here is that this will break the `.keywordcode` and `.constantcode` commands, since we change from an owning string (`std::string`) to a non-owning string (`const char*`). I don't think these commands are really used, so perhaps better to just remove them unless someone objects.

Reply:
Really? If you give `.keywordcode` as `"code"`  it should work. Or at least I think so. I will look next if it's possible to remove `termios.h`. Then it would maybe be possible to use same code in Windows.

Reply:
Have you tried this? I suspect the program will crash because you are holding a reference to a string that will be deleted. std::string actually owns the data which is how this worked before.

Reply:
I haven't tried that but with `.resetcolors` I'll do `keyword = "\033[32m\033[1m";` and It works nicely.

Reply:
It's working with `strdup`.

Reply:
> I haven't tried that but with .resetcolors I'll do keyword = "\033[32m\033[1m"; and It works nicely.

That is because that is a string constant.

> It's working with strdup

That will cause memory leaks which are not desirable. Perhaps check if you can replace `keyword`, etc, with functions like this:

```cpp
string &keyword() {
	static const string kw;
	return kw;
}
```

That will allow you to use a `std::string` object, while (hopefully) fixing the compilation on zig.

Reply:
Alternatively, a better solution might be to have a static buffer (e.g. `char keyword_storage[20]`), then do a copy of the parameter in `.keywordcode` to that static buffer.